internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -439,9 +527,124 @@ func validateSecretConfig(secretConfig SecretConfig) []string {
 
-	if secretConfig.IntegrationName == "" {
-		missingKeys = append(missingKeys, "integration_name")
+	return missingKeys
+}
+
+var availableRotatingSecretProviders = map[string]map[string]any{
+	string(integrations.Twilio):       twilioRotatingSecretConfig,
+	string(integrations.MongoDBAtlas): mongoDBAtlasRotatingSecretConfig,
+	string(integrations.AWS):          awsRotatingSecretConfig,
+	string(integrations.GCP):          gcpRotatingSecretConfig,
+}
+
+var availableDynamicSecretProviders = map[string]map[string]any{
+	string(integrations.AWS): awsDynamicSecretConfig,
+	string(integrations.GCP): gcpDynamicSecretConfig,
+}
+
+func promptUserForConfig(opts *CreateOpts) (SecretConfig, secretConfigInternal, error) {
+	var (
+		config         SecretConfig
+		internalConfig secretConfigInternal
+		err            error
+		providerFields map[string]map[string]any
+		items          []string
+	)
+
+	if !opts.IO.CanPrompt() {
+		return config, internalConfig, fmt.Errorf("unable to create secret interactively")
 	}
 
-	return missingKeys
+	if opts.Type == secretTypeDynamic {
+		providerFields = availableDynamicSecretProviders
+		items = maps.Keys(providerFields)
+	} else if opts.Type == secretTypeRotating {
+		providerFields = availableRotatingSecretProviders
+		items = maps.Keys(providerFields)
+	}
+
+	providerPrompt := promptui.Select{
+		Label:  "Please select the provider you would like to configure",
+		Items:  items,
+		Stdin:  io.NopCloser(opts.IO.In()),
+		Stdout: iostreams.NopWriteCloser(opts.IO.Err()),
+	}
+
+	_, provider, err := providerPrompt.Run()
+	if err != nil {
+		return config, internalConfig, fmt.Errorf("provider selection prompt failed: %w", err)
+	}
+	config.Type = integrations.IntegrationType(provider)
+
+	fieldValues, err := populateFieldValues(providerFields[provider], opts)
+	if err != nil {
+		return config, internalConfig, err
+	}
+
+	internalConfig.Details = fieldValues
+	return config, internalConfig, err
+}
+
+func populateFieldValues(fieldMap map[string]any, opts *CreateOpts) (map[string]any, error) {
+	var fieldPrompt promptui.Prompt
+	fieldsMap := make(map[string]any)
+
+	for _, field := range maps.Keys(fieldMap) {
+		value := fieldMap[field]
+		switch reflect.ValueOf(value).Kind() {
+		case reflect.String:
+			fieldPrompt = promptui.Prompt{
+				Label: field,
+				Mask:  '*',
+			}
+
+			input, err := fieldPrompt.Run()
+			if err != nil {
+				return fieldsMap, fmt.Errorf("prompt for field %s failed: %w", field, err)
+			}
+
+			fieldsMap[field] = input
+		case reflect.Map:
+			nestedMap, err := populateFieldValues(value.(map[string]any), opts)
+			if err != nil {
+				return nil, err
+			}
+			fieldsMap[field] = nestedMap
+		case reflect.Slice:
+			var (
+				sliceItems []map[string]any
+			)
+			nestedMap := make(map[string]any)
+			valueSlice := value.([]map[string]string)
+
+			for proceed := true; proceed; proceed = promptForAdditionalField(field, opts) {
+
+				for _, nestedField := range maps.Keys(valueSlice[0]) {
```

Comment:
also generally a little curious why you're not doing the recursive approach when the type is a map here as well, if this is a slice of maps

Reply:
> i think i'm a little confused on the logic behind maps.Keys(valueSlice[0]) here, would you mind dropping a comment about it?
I can definitely drop a comment here. In short, since the `valueSlice` (slice of maps) is hardcoded in the provider field templates, we know there is only a map at the first index of the slice. So instead of iterating through the slice, let's just grab the first index of the slice and iterative over the map keys at that index.

> also generally a little curious why you're not doing the recursive approach when the type is a map here as well, if this is a slice of maps
Good question. Only because the [prompt](https://github.com/hashicorp/hcp/pull/167/files#diff-7f92b17a95650abeca3ec432e8bcf6859f19709d1bdb6b3687fb5d704940a4ecR624) is a bit different when it's a slice, but let me see if I reconcile that for it to also be recursive.

Reply:
ahhh, didn't realize `valueSlice` would only have 1 elem there if its a slice of maps, thanks for clarifying!